src/explorers.ts

```diff
@@ -29,7 +31,9 @@ interface Explorer {
 export type ExplorerType = keyof typeof explorers
 export type ExplorerLinkType = 'account' | 'transaction'
 
-export const generateExplorerUrl = (explorer: "solanaExplorer" | "solscan" | "solanaFm", signatureOrAccount: string, type: ExplorerLinkType): string => {
+export const generateExplorerUrl = async (explorer: "solanaExplorer" | "solscan" | "solanaFm", rpcUrl: string, signatureOrAccount: string, type: ExplorerLinkType): Promise<string> => {
+
+    const chain = await checkRpcChain(rpcUrl)
```

Comment:
chain should be passed into this method and precalculated during the context creation and stored in the context IMO. It would mean that `generateExplorerUrl` doesn't need to be async anymore and you can repeatedly call the function for lists of things (e.g. assets created or soemthing) and not require continually calling the RPC for genesis hash